okhttp/src/main/java/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -227,6 +227,8 @@ class RealConnection(
       throw RouteException(ProtocolException(
           "Too many tunnel connections attempted: $MAX_TUNNEL_ATTEMPTS"))
     }
+
+    idleAtNs = System.nanoTime()
```

Comment:
Assume this triggers the behaviour change.

Reply:
Yep.